### PR TITLE
Fix: Add bookmark by shift+drag #390

### PR DESCRIPTION
**Pull request by IpankajI**
This branch has all changes of one previousy branch
<fix_csv_export_shift_drag> which was based on <develop>
,but this one is based on <new_dir_structure>.

        CSV exported by holding shift key down and dragging mouse
had no info for samples because here new object of <PeakGroup>
is generated but <samples> were not assigned to this group.
        Now all selected samples has been assigned to group.

One new function <setSelectedSamples> has been introduced into
<PeakGroup.cpp> file which will assign all selected samples to it's
<samples> varible and all manual sample selection and assignment
has been removed from all other places

Issue:#390

### CODE REVIEW

File: src/core/libmaven/PeakGroup.h [hunk 1]
```diff
@@ -540,5 +540,7 @@ class PeakGroup{
                             int qualityWeight,
                             int intensityWeight,
                             int deltaRTWeight);
+
+        void setSelectedSamples(vector<mzSample*> vsamples);
```
Comment (sahil21) [inline src/core/libmaven/PeakGroup.h]: Documentation of this function + unit tests.

File: src/gui/mzroll/eicwidget.cpp [hunk 2]
```diff
@@ -151,7 +151,7 @@ void EicWidget::integrateRegion(float rtmin, float rtmax) {
 	eicParameters->_integratedGroup.minQuality = settings->value("minQuality").toDouble();
 	eicParameters->_integratedGroup.compound = eicParameters->_slice.compound;
 	eicParameters->_integratedGroup.srmId = eicParameters->_slice.srmId;
-
+	eicParameters->_integratedGroup.setSelectedSamples(getMainWindow()->samples);
```
Comment (sahil21) [inline src/gui/mzroll/eicwidget.cpp]: Update documentation of this function

File: src/core/libmaven/EIC.cpp [hunk 3]
```diff
@@ -729,6 +729,10 @@ vector<PeakGroup> EIC::groupPeaks(vector<EIC *> &eics,
                                   bool useOverlap,
                                   double minSignalBaselineDifference)
 {
+    vector<mzSample*> samples;
```
Comment (sahil21) [inline src/core/libmaven/EIC.cpp]: Update document of groupPeaks

File: src/gui/mzroll/eicwidget.cpp (hunk 2 above)
Comment (IpankajI) [reply src/gui/mzroll/eicwidget.cpp]: @sahil21 done

**Pull request merged**